### ISSUE TITLE
Determine argument type automatically from default

### ISSUE THREAD

**Issue by jdidion** [micheles/plac#22]
```
if an argument data type is not provided explicitly but a default value is:
    if default is not a tuple:
        argtype = type(default)
    elif all(type(val) == type(default[0]) for val in default[1:]):
        argtype = type(default[0])
```

**Comment by micheles** [comment]
Care to expand on that? What would be your use case?

**Comment by jdidion** [comment]
Sure, sorry. It’s a common use case for me: I want to specify a non-string default value for a positional argument, and have the argument parsed as the correct data type, without having to provide unnecessary/redundant information in the attribute. For example,

this:

def main(size: “The size of the thing” = 10):
   …

is much easier to write than

def main(size: (“The size of the thing”, “positional”, None, int) = 10):
   ...

In my mind, the first is also easier to understand, as I’d assume that a parameter with an integer default always expects an integer argument.

Another alternative would be to allow the attribute to be a dict rather than a tuple, so I could specify only fields I need:

def main(size: dict(help=“The size of the thing”, type=int) = 10):
    …

Which is at least easier to parse, since now I don’t have to know the order of the fields in the attribute tuple to understand it, but still, in my opinion, more verbose than necessary.

**Comment by micheles** [comment]
This is actually a good idea. I will keep it in mind, but I do not known when I will have time to work on it.

**Comment by jdidion** [comment]
Thanks. If I have time I will submit a pull request.

**Comment by micheles** [comment]
This was actually implemented in plac 1.0, one year ago, but I forgot to close the issue.
